tiledb/api/c_api/datatype/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+#
+# tiledb/api/c_api/datatype/CMakeLists.txt
+#
+# The MIT License
+#
+# Copyright (c) 2022 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+include(common NO_POLICY_SCOPE)
+
+list(APPEND SOURCES
+  datatype_api.cc
+)
+gather_sources(${SOURCES})
+
+#
+# Object library for other units to depend upon
+#
+add_library(capi_datatype OBJECT ${SOURCES})
+target_link_libraries(capi_datatype PUBLIC export)
+target_link_libraries(capi_datatype PUBLIC constants $<TARGET_OBJECTS:constants>)
+
+#
+# Test-compile of object library ensures link-completeness
+#
+add_executable(compile_capi_datatype EXCLUDE_FROM_ALL)
+target_sources(compile_capi_datatype PRIVATE test/compile_capi_datatype_main.cc)
+target_link_libraries(compile_capi_datatype PRIVATE capi_datatype)
+target_link_libraries(compile_capi_datatype PRIVATE baseline)
```

Comment:
Compile-tests are only allowed a single object library dependency. If they require something else, then they don't link independently. The new `object_library` environment enforces this by generating the compile test from only (1) the object library being defined and (2) a `compile_*_main.cc` file with a non-reconfigurable name. As it stands, this object library can't be converted to the new environment, because it would fail to link.

Reply:
Makes sense. I've update the baseline dependency on each object library and removed all downstream dependencies on baseline for the compile and unit tests.